ci/wezterm-shellextension/WezTermContextMenu/src/dllmain.cpp

```diff
@@ -0,0 +1,234 @@
+            }
+        }
+        return E_FAIL;
+    }
+
+
+
+	IFACEMETHODIMP GetToolTip(_In_opt_ IShellItemArray*, _Outptr_result_nullonfailure_ PWSTR* infoTip) { *infoTip = nullptr; return E_NOTIMPL; }
+	IFACEMETHODIMP GetCanonicalName(_Out_ GUID* guidCommandName) { *guidCommandName = GUID_NULL; return S_OK; }
+/**
+* @brief GetState retrieves the state of the context menu item.
+*
+* This method determines the state of the context menu item based on the provided selection and system conditions.
+*
+* @param selection The selected items in the shell.
+* @param okToBeSlow Indicates whether it's acceptable to be slow.
+* @param cmdState Output parameter to receive the state of the context menu item (EXPCMDSTATE).
+* @return HRESULT indicating success (S_OK) or failure.
+*/
+    IFACEMETHODIMP GetState(_In_opt_ IShellItemArray* selection, _In_ BOOL okToBeSlow, _Out_ EXPCMDSTATE* cmdState)
+    {
+        *cmdState = ECS_ENABLED;
+        return S_OK;
+    }
+/**
+* @brief Invoke executes the context menu item action.
+*
+* This method is called to perform the action associated with the context menu item. It displays debug messages
+* for invalid arguments or if there are no items to process. It retrieves the file path of the selected item,
+* extracts information about the file.
+*
+* @param selection The selected items in the shell.
+* @param bindContext The bind context.
+* @return HRESULT indicating success (S_OK) or failure.
+*/
+
+    IFACEMETHODIMP Invoke(_In_opt_ IShellItemArray* selection, _In_opt_ IBindCtx*) noexcept try {
+        if (!selection) {
+            // Debug message
+            MessageBox(nullptr, L"Selection is nil. Please select a valid item.", L"WezTerm Shell Extension", MB_OK);
+            return E_INVALIDARG;
+        }
+
+
+        DWORD count;
+        RETURN_IF_FAILED(selection->GetCount(&count));
+
+        if (count == 0) {
+            // Debug message
+            MessageBox(nullptr, L"No items to process", L"WezTerm Shell Extension", MB_OK);
+            return S_OK; // No items to process
+        }
+
+        ComPtr<IShellItem> item;
+        RETURN_IF_FAILED(selection->GetItemAt(0, &item));
+
+        // Get the absolute path of the parent directory of the selected item
+        PWSTR parentDirPath;
+        RETURN_IF_FAILED(item->GetDisplayName(SIGDN_DESKTOPABSOLUTEPARSING, &parentDirPath));
+        wil::unique_cotaskmem_string parentDirPathCleanup(parentDirPath);
+
+        // Debug: Display the absolute directory path in a message box
+        std::wstring message = L"Directory path: " + std::wstring(parentDirPath);
+
+        // Get the directory containing the DLL as the base path for wezterm-gui.exe
+        wchar_t dllDirectory[MAX_PATH];
+        GetModuleFileName(g_hModule, dllDirectory, MAX_PATH);
+        PathRemoveFileSpec(dllDirectory);
+
+        // Construct the full path to wezterm-gui.exe
+        std::wstring wezExePath = std::wstring(dllDirectory) + L"\\wezterm-gui.exe";
```

Comment:
this is the duplicate code with that in the GetIcon function; it needs to be factored out